mshadow/tensor.h

```diff
@@ -299,6 +299,7 @@ inline Shape<4> ConvertLayout(const Shape<4>& src, int src_layout, int dst_layou
     break;
   default:
     LOG(FATAL) << "Invalid layout for 4d shape " << src_layout;
+    dst = src;  // fixes compiler warning
```

Comment:
Interestingly no. It doesn't raise any warning for the functions above and below this function, which seem to work similarly